pkg/auth/dependency/userprofile/store.go

```diff
@@ -0,0 +1,89 @@
+package userprofile
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/skygeario/skygear-server/pkg/core/db"
+)
+
+type storeImpl struct {
+	sqlBuilder  db.SQLBuilder
+	sqlExecutor db.SQLExecutor
+	logger      *logrus.Entry
+}
+
+func newUserProfileStore(builder db.SQLBuilder, executor db.SQLExecutor, logger *logrus.Entry) *storeImpl {
+	return &storeImpl{
+		sqlBuilder:  builder,
+		sqlExecutor: executor,
+		logger:      logger,
+	}
+}
+
+func (u storeImpl) CreateUserProfile(userID string, userProfile map[string]interface{}) (err error) {
+	now := time.Now().UTC()
```

Comment:
The inline use of time.Now().UTC is not testable.

After the discussion on, in skygear project we adopted this: https://github.com/SkygearIO/skygear-server/pull/518/files